src/clr-angular/timeline/timeline-step.spec.ts

```diff
@@ -0,0 +1,221 @@
+
+        step.state = ClrTimelineStepState.CURRENT;
+        expect(step.iconAriaLabel).toBe('Current');
+
+        step.state = ClrTimelineStepState.PROCESSING;
+        expect(step.iconAriaLabel).toBe('In progress');
+
+        step.state = ClrTimelineStepState.SUCCESS;
+        expect(step.iconAriaLabel).toBe('Completed');
+
+        step.state = ClrTimelineStepState.ERROR;
+        expect(step.iconAriaLabel).toBe('Error');
+      });
+
+      it('set the aria-current', () => {
+        step.state = ClrTimelineStepState.NOT_STARTED;
+        expect(step.iconAriaCurrent).toBe(false);
+
+        step.state = ClrTimelineStepState.CURRENT;
+        expect(step.iconAriaCurrent).toBe(true);
+
+        step.state = ClrTimelineStepState.PROCESSING;
+        expect(step.iconAriaCurrent).toBe(false);
+
+        step.state = ClrTimelineStepState.SUCCESS;
+        expect(step.iconAriaCurrent).toBe(false);
+
+        step.state = ClrTimelineStepState.ERROR;
+        expect(step.iconAriaCurrent).toBe(false);
+      });
+    });
+
+    describe('Template API', () => {
+      let fixture: ComponentFixture<TestTimelineStep>;
+      let hostComponent: TestTimelineStep;
+      let debugElement: DebugElement;
+      let nativeComponent: ClrTimelineStep;
+
+      beforeEach(() => {
+        TestBed.configureTestingModule({
+          declarations: [ClrTimelineStep, TestTimelineStep],
+          imports: [ClrIconModule, ClrSpinnerModule],
+        });
+
+        fixture = TestBed.createComponent(TestTimelineStep);
+        fixture.detectChanges();
+
+        hostComponent = fixture.componentInstance;
+        debugElement = fixture.debugElement;
+        nativeComponent = hostComponent.step;
+      });
+
+      it('accepts a [clrState] input', () => {
+        hostComponent.state = ClrTimelineStepState.NOT_STARTED;
+        fixture.detectChanges();
+        expect(nativeComponent.state).toBe(hostComponent.state);
+
+        hostComponent.state = ClrTimelineStepState.CURRENT;
+        fixture.detectChanges();
+        expect(nativeComponent.state).toBe(hostComponent.state);
+
+        hostComponent.state = ClrTimelineStepState.SUCCESS;
+        fixture.detectChanges();
+        expect(nativeComponent.state).toBe(hostComponent.state);
+
+        hostComponent.state = ClrTimelineStepState.ERROR;
+        fixture.detectChanges();
+        expect(nativeComponent.state).toBe(hostComponent.state);
+
+        hostComponent.state = ClrTimelineStepState.PROCESSING;
+        fixture.detectChanges();
+        expect(nativeComponent.state).toBe(hostComponent.state);
+      });
+
+      it('renders icon when not processing', () => {
+        hostComponent.state = ClrTimelineStepState.CURRENT;
+        fixture.detectChanges();
+
+        const icon = debugElement.query(By.css('clr-icon'));
+        const spinner = debugElement.query(By.css('clr-spinner'));
+
+        expect(icon).not.toBeNull();
+        expect(spinner).toBeNull();
```

Comment:
I think we should cover the other non processing states here in addition to `ClrTimelineStepState.CURRENT`.